### DISCUSSION TITLE
Refrandom

### DISCUSSION THREAD

**Discussion by azinbr** [WeAreMahsaAmini/Core#275] [Ideas]
I suggest that you create a section on the site for a safe referendum
that everyone can vote only once
And anyone can see the results of the votes until now

**Comment by ashkist** [comment]
cross-posting from discord (I mentioned this in a wrong thread)

If Party A like Anonymous could have access to the national documents, they can validate it with proper hashing. Party B which would be the vote counter service like us would count the votes independent of knowing who is behind them.

In case of objection, people have the freedom to publish their vote to show if it's counted or not. (as all freedom fighters have done in history, this time we make sure all of them are heard and safe)

I also was digging the 2 links below to make the idea work. 

https://crypto.stanford.edu/pbc/notes/crypto/voting.html
https://www.mdpi.com/2073-8994/14/5/858/pdf

BTW, by work I mean it's theoretically feasible with the current infrastructure and tools people have. 

The reason we didn't act on it was that there needs to be enough trust and willingness from the people. But I'm happy to discuss about it and help anyone who wants to be made (and of course, I would be there too if I'm still around 🙌🏻 ).
Keep in mind, A Review of Cryptographic Electronic Voting is published this year.

Also this year we had The Network State by Balaji (TheNetworkState.com) which describes a social movement like this. A Regulation State that doesn't need ownership of power to rise to power.

We're at the very beginning of the era that this is possible, and we can do it with minimum resources and maximum reliably. At least in comparison to any elections before, globally.

**Reply by ashkist** [reply]
I think we still need some time to get to this. Maybe this would be helpful as a failed draft for a better appraoch:

```mermaid
sequenceDiagram

Participant ru as Real User
Participant anonKey as Anonymous Asymmetric Key
Participant anonVendor as Anonymous Vote Ticket Vendor
Participant vdvc as Vote Day Voting Centre

ru ->> anonKey: 1. Send the Anonymous Key
anonKey ->> ru : 2.

ru ->> anonVendor: 3. Send Secret ID
ru ->> anonVendor: 4. Password
anonVendor ->> ru: 5. Send **FreeVote Ticket**

ru ->> vdvc: 6. Vote
vdvc ->> ru: 7. Sends Reference Code
```

1. Create a random **Private Key** and send the register the **Public Key** as the **Anonymous Key**.
2. Get validation of **Public Key**.
3. Send Secret ID: Hash of **National ID** attributes and get toke validation by an Anonymous Vendor or any other trusted vendor
4. Set a password on the Key (optional + extra)
5. Get Ticket: Send voting valid **National Voting Ticket** to the user and publish hash for general validation of mixers.
6. Vote: Send the vote signed by the key and middle mixers (to keep the citizen anonymous)
7. Sends Reference Code: (not sure) The user should be able to check if his/her vote is counted.

Note: If the user wants to make sure the vote is counted, they could make it public (since it's usable only once and theoretically possible)

**Reply by ashkist** [reply]
A **Feasible Personal Data** for the Private Key which is sent to **Anonymous Vote Ticket Vendor** could be a `salt` with the addition of the mix of values below (1 or more):

1. National ID Code
2. The Code behind National ID
3. Barcode behind National ID

The approach above will need additional support for people who only own a **Passport** or **Birth Certificate**.
